### PR TITLE
[GEOT-6938] Image mosaic: avoid sorting on CRS when doing a 'dry run' check after finding no granules in the target area

### PR DESCRIPTION
**Pull request by aaime**
[GEOT-6938](https://osgeo-org.atlassian.net/browse/GEOT-6938) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=geotools&utm_campaign=badge_info)

See tickets for details

# Checklist

Reviewing is a process done by project maintainers, **mostly on a volunteer basis** (thus limited in time). We need to keep the review overhead as small as possible, and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geotools/geotools/blob/main/CONTRIBUTING.md)
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the `main` branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.
- [x] The changes are not causing two modules to share the same Java packages (to avoid [Java 9+ split package](http://tutorials.jenkov.com/java/modules.html#split-packages-not-allowed) issues)
- [x] The changes are not breaking the build in downstream projects using SNAPSHOT dependencies, GeoWebCache and GeoServer (there is an automatic PR check verifying this, check this when it turns green).

The following are required only for core and extension modules (they are welcomed, but not required, for unsupported modules):

- [x] There is an issue in [Jira](https://osgeo-org.atlassian.net/projects/GEOT) describing the bug/task/new feature (a notable exemptions is, changes not visible to end users). The ticket is for the GeoTools project, if the issue was found elsewhere it's a good practice to link to the origin ticket/issue.
- [x] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [x] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message(s) must be in the form "[GEOT-XYZW] Title of the Jira ticket"
- [x] New unit tests have been added covering the changes
- [x] This PR passes all existing unit tests (test results will be reported by Continuous Integration after opening this PR)
- [x] This PR passes the [QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html) (QA checks results will be reported by Continuous Integration after opening this PR)
- [ ] Documentation has been updated accordingly.

Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or not applicable.

### CODE REVIEW

File: modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/catalog/GranuleCatalogFactory.java [hunk 1]
```diff
@@ -86,9 +86,18 @@ public static GranuleCatalog createGranuleCatalog(
...
+            try {
+                store =
+                        Optional.ofNullable(
+                                        (DataStore) ((Wrapper) store).unwrap(JDBCDataStore.class))
+                                .orElse(store);
+            } catch (IllegalArgumentException e) {
+                LOGGER.log(
+                        Level.FINER,
+                        "The store is a wrapper but does not wrap a JDBCDataStore "
+                                + "(not a problem per se, just a note",
```
Comment (dromagnoli) [inline modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/catalog/GranuleCatalogFactory.java:98]: I would close the ")" :)

File: modules/plugin/imagemosaic/src/test/java/org/geotools/gce/imagemosaic/HeterogenousCRSTest.java [hunk 2]
```diff
@@ -958,4 +967,83 @@ public void testNativeEnvelope() throws Exception {
...
+
+        // record queries only during the read
+        recordQueries.set(true);
+        GridCoverage2D coverage = repoReader.read(new GeneralParameterValue[] {ggp, filter});
+        recordQueries.set(false);
+        if (coverage != null) coverage.dispose(true);
+        repoReader.dispose();
+        repository.dataStore(TEST_STORE).dispose();
+
+        // check the last query did not have the sorting, but the second to last, did not
```
Comment (dromagnoli) [inline modules/plugin/imagemosaic/src/test/java/org/geotools/gce/imagemosaic/HeterogenousCRSTest.java:1043]: I think there is a small TYPO here. 
the second to last has sorting so probably the statement should end with "but the second to last did it".

**Review by dromagnoli — APPROVED** [review]
Looks good to me beside a couple of small typo

File: modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/catalog/GranuleCatalogFactory.java (hunk 1 above)
Comment (aaime) [reply modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/catalog/GranuleCatalogFactory.java:98]: Yep, fixed

File: modules/plugin/imagemosaic/src/test/java/org/geotools/gce/imagemosaic/HeterogenousCRSTest.java (hunk 2 above)
Comment (aaime) [reply modules/plugin/imagemosaic/src/test/java/org/geotools/gce/imagemosaic/HeterogenousCRSTest.java]: Improved explanation and improved assertions while I was at it.

**Pull request merged**